lib/src/models.dart

```diff
@@ -204,19 +204,21 @@ class Arrow implements Shape {
 /// A piece shape that can be drawn on the board.
 @immutable
 class PieceShape implements Shape {
-  final Color color;
-  final Role role;
+  final Piece piece;
   final Square orig;
+  final PieceAssets pieceAssets;
+  final double opacity;
   @override
   final double scale;
 
   /// Creates a new [PieceShape] with the provided values.
   ///
   /// The [scale] must be between 0.0 and 1.0.
   const PieceShape({
-    required this.color,
-    required this.role,
+    required this.piece,
     required this.orig,
+    required this.pieceAssets,
+    this.opacity = 1.0,
```

Comment:
Since it has to be distinguished from regular pieces, the default should not be 1.0.

0.5 should be fine.